### PR TITLE
GH-37022: [CI][Java] fix maven 3.9.3 causing docker build failure

### LINKED ISSUE #37022: [CI][Java] Maven 3.9.3 causing Docker build failure
### Describe the bug, including details regarding any error messages, version, and platform.

running:

archery docker run java-jni-manylinux-2014

causing

```
java-jni-manylinux-201x.dockerfile:38
--------------------
  37 |     ARG maven=3.9.3
  38 | >>> RUN yum install -y java-$java-openjdk-devel && \
  39 | >>>       yum clean all && \
  40 | >>>       curl https://dlcdn.apache.org/maven/maven-3/${maven}/binaries/apache-maven-${maven}-bin.tar.gz | \
  41 | >>>         tar xfz - -C /usr/local && \
  42 | >>>       ln -s /usr/local/apache-maven-${maven}/bin/mvn /usr/local/bin
  43 |
--------------------
ERROR: failed to solve: process "/bin/bash -i -c yum install -y java-$java-openjdk-devel &&       yum clean all &&       curl https://dlcdn.apache.org/maven/maven-3/${maven}/binaries/apache-maven-${maven}-bin.tar.gz |         tar xfz - -C /usr/local &&       ln -s /usr/local/apache-maven-${maven}/bin/mvn /usr/local/bin" did not complete successfully: exit code: 2
Service 'java-jni-manylinux-2014' failed to build
Error: `docker-compose --file /root/zhenw/arrow/docker-compose.yml build --build-arg BUILDKIT_INLINE_CACHE=1 java-jni-manylinux-2014` exited with a non-zero exit code 1, see the process log above.
```

it seems the link for maven version 3.5.4 is no longer available

```
curl https://dlcdn.apache.org/maven/maven-3/3.5.4/binaries/apache-maven-3.5.4-bin.tar.gz
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL was not found on this server.</p>
</body></html>
```

### Component(s)

Release

### PR DESCRIPTION
**Pull request by zinking**
### Rationale for this change

docker build failed due to in correct maven 3.9.3 artifact.

### Are there any user-facing changes?

* Closes: #37022

### CODE REVIEW

File: ci/docker/java-jni-manylinux-201x.dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ RUN vcpkg install \
 
 # Install Java
 ARG java=1.8.0
-ARG maven=3.9.3
+ARG maven=3.9.4
 RUN yum install -y java-$java-openjdk-devel && \
```
Comment (zinking) [inline ci/docker/java-jni-manylinux-201x.dockerfile:38]: I was thinking of splitting this single line into multiple lines to make errors explicit.

File: ci/docker/java-jni-manylinux-201x.dockerfile [hunk 2]
```diff
@@ -34,7 +34,7 @@ RUN vcpkg install \
...
       yum clean all && \
       curl https://dlcdn.apache.org/maven/maven-3/${maven}/binaries/apache-maven-${maven}-bin.tar.gz | \
```
Comment (kou) [inline ci/docker/java-jni-manylinux-201x.dockerfile:40]: Could you try the following URL with 3.9.3?

```suggestion
      curl https://www.apache.org/dyn/closer.lua?action=download&filename=maven/maven-3/${maven}/binaries/apache-maven-${maven}-bin.tar.gz | \
```

Comment (zinking) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]: hmm, let me try later tomorrow. but I thought it was not a big deal.

Comment (kou) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]: Then could you use the URL and revert the Maven bump?

The `.../dyn/closer.lua...` URL is the official URL.

Comment (zinking) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]:
```
------
 > [ 6/11] RUN curl https://www.apache.org/dyn/closer.lua?action=download&filename=maven/maven-3/3.9.3/binaries/apache-maven-3.9.3-bin.tar.gz |         tar xfz - -C /usr/local:
0.136 bash: cannot set terminal process group (-1): Inappropriate ioctl for device
0.136 bash: no job control in this shell
0.139 [1] 9
0.141
0.141 gzip: stdin: unexpected end of file
0.142 tar: Child returned status 1
0.142 tar: Error is not recoverable: exiting now
```

this is a bit weird, things works fine when I do it from my mac. but on centos it complains with above error.

Comment (kou) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]: Could you try `curl --location https://...`?

Comment (zinking) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]:
```
curl --location https://www.apache.org/dyn/closer.lua\?action\=download\&filename\=maven/maven-3/$\{maven\}/binaries/apache-maven-$\{maven\}-bin.tar.gz | \
pipe>         tar xfz - -C /usr/local
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    74  100    74    0     0     83      0 --:--:-- --:--:-- --:--:--    83

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```

but if I down load and run tar zxf against the file, seems worked.

Comment (kou) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]: Could you show debug log by `curl --location -v https://...`?

Comment (zinking) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]: this is the one with curl out piped to tar zxf 
```
curl -v --location https://www.apache.org/dyn/closer.lua\?action\=download\&filename\=maven/maven-3/$\{maven\}/binaries/apache-maven-$\{maven\}-bin.tar.gz | \
pipe>         tar xfz - -C /usr/local
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to www.apache.org port 443 (#0)
*   Trying 151.101.2.132...
* Connected to www.apache.org (151.101.2.132) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
  0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0* NSS error -12263 (SSL_ERROR_RX_RECORD_TOO_LONG)
* SSL received a record that exceeded the maximum permissible length.
  0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0
* Closing connection 0
curl: (35) SSL received a record that exceeded the maximum permissible length.

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```

this is the one downloaded to a target file first
```
curl -v --location https://www.apache.org/dyn/closer.lua\?action\=download\&filename\=maven/maven-3/3.9.3/binaries/apache-maven-3.9.3-bin.tar.gz -o apache-maven-3.9.3-bin.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to www.apache.org port 443 (#0)
*   Trying 151.101.2.132...
* Connected to www.apache.org (151.101.2.132) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* SSL connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate:
* 	subject: CN=*.apache.org
* 	start date: Jun 12 16:01:34 2023 GMT
* 	expire date: Sep 10 16:01:33 2023 GMT
* 	common name: *.apache.org
* 	issuer: CN=R3,O=Let's Encrypt,C=US
> GET /dyn/closer.lua?action=download&filename=maven/maven-3/3.9.3/binaries/apache-maven-3.9.3-bin.tar.gz HTTP/1.1
> User-Agent: curl/7.29.0
> Host: www.apache.org
> Accept: */*
>
< HTTP/1.1 302 Found
< Connection: keep-alive
< Content-Length: 0
< Server: Apache
< Cache-Control: private, max-age=3600
< Location: https://archive.apache.org/dist/maven/maven-3/3.9.3/binaries/apache-maven-3.9.3-bin.tar.gz
< Expires: Tue, 08 Aug 2023 04:17:27 GMT
< Content-Security-Policy: default-src 'self' 'unsafe-inline' https://www.apachecon.com/ https://www.google.com/cse/ https://cse.google.com/ https://www.googleapis.com/generate_204 http://*.google.com/generate_204 https://afs.googlesyndication.com/ https://csp.withgoogle.com/ https://www.google.com/images/ https://ssl.gstatic.com/ui/ https://docs.google.com/forms/ https://www.youtube.com/embed/; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://cse.google.com/ http://cse.google.com/adsense/search/async-ads.js https://www.google.com/cse/ https://partner.googleadservices.com/; style-src 'self' 'unsafe-inline' https://www.google.com/cse/; frame-ancestors 'none';
< Strict-Transport-Security: max-age=31536000; preload
< Accept-Ranges: bytes
< Via: 1.1 varnish, 1.1 varnish
< Date: Tue, 08 Aug 2023 03:17:27 GMT
< X-Served-By: cache-hel1410020-HEL, cache-tyo11949-TYO
< X-Cache: MISS, MISS
< X-Cache-Hits: 0, 0
< X-Timer: S1691464647.113735,VS0,VE270
<
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host www.apache.org left intact
* Issue another request to this URL: 'https://archive.apache.org/dist/maven/maven-3/3.9.3/binaries/apache-maven-3.9.3-bin.tar.gz'
* About to connect() to archive.apache.org port 443 (#1)
*   Trying 65.108.204.189...
* Connected to archive.apache.org (65.108.204.189) port 443 (#1)
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0* SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate:
* 	subject: CN=*.apache.org
* 	start date: Jul 25 00:00:00 2023 GMT
* 	expire date: Aug 24 23:59:59 2024 GMT
* 	common name: *.apache.org
* 	issuer: CN=Sectigo RSA Domain Validation Secure Server CA,O=Sectigo Limited,L=Salford,ST=Greater Manchester,C=GB
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0> GET /dist/maven/maven-3/3.9.3/binaries/apache-maven-3.9.3-bin.tar.gz HTTP/1.1
> User-Agent: curl/7.29.0
> Host: archive.apache.org
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Tue, 08 Aug 2023 03:17:28 GMT
< Server: Apache
< Vary: Slow,Glacial
< Last-Modified: Mon, 26 Jun 2023 11:09:31 GMT
< ETag: "8d4679-5ff06628e5bc7"
< Accept-Ranges: bytes
< Content-Length: 9258617
< Content-Type: application/x-gzip
<
{ [data not shown]
100 9041k  100 9041k    0     0   252k      0  0:00:35  0:00:35 --:--:--  508k
* Connection #1 to host archive.apache.org left intact
```

Comment (kou) [reply ci/docker/java-jni-manylinux-201x.dockerfile:40]: Thanks. It seems that  it's a (old?) `curl` problem on CentOS 7.

Could you use the following with Maven 3.9.3? 

```suggestion
      curl --location https://www.apache.org/dyn/closer.lua?action=download&filename=maven/maven-3/${maven}/binaries/apache-maven-${maven}-bin.tar.gz | \
```

We don't use CentOS 7 on our CI. So we will not have the problem on our CI. (You can use macOS not CentOS 7 for this, right?)

**Pull request closed without merging**